bemani/backend/jubeat/clan.py

```diff
@@ -1809,7 +1816,7 @@ def unformat_profile(self, userid: UserID, request: Node, oldprofile: ValidatedD
                     if flags & bit > 0:
                         medal = max(medal, mapping[bit])
 
-                self.update_score(userid, timestamp, songid, chart, points, medal, combo, ghost, stats)
+                self.update_score(userid, timestamp, songid, chart, points, medal, combo, ghost, stats, hard_mode=is_hard_mode)
```

Comment:
This is where base DB charts should be calculated based on the game, instead of passing hard_mode down. That's an implementation detail that the base game shouldn't care about.